### DISCUSSION TITLE
show .gif or LivePhotos

### DISCUSSION THREAD

**Discussion by vinyli85** [pi3d/pi3d_demos#37] [Q&A]
I managed to show .gif-Files but only as a steady image.
Is there a way to show gif or LivePhotos (like taken on iOS) or is this not a function of `pi3d` but of `Pillow`?

**Comment by paddywwoof** [comment]
I think this question came up before, I will see if I can dig out what I did then. You can open a multiframe gif image using pillow (as you found out, that's what pi3d uses) but to get the additional frames you need to seek between them along the lines of (line 74 or so in PictureFrame
```python
...
    im = Image.open(fname)
    im.seek(f_num)
...
```
However for this to work you would have to initially get the number of frames (easy enough using ```im.is_animated``` and ```im.n_frames```) and you would need to increment the frame count each frame and stop the fname from incrementing until you got to the end of the animation.

Additionally the speed of the the RPi GPU system used by pi3d in PictureFrame is going to be rather limited, certainly for high resolution pictures (i.e. bigger than 640x480). Obviously if you wanted to mix fast fast frame changing withing a gif but slow frame changing from one image to the next you would have to figure out additional mechanisms to ensure that happened (Probably the animation of the gif could only start after the fading had finished)

**Reply by vinyli85** [reply]
Thanks for your answer, Paddy. It sounds like it is a bit too much for the GPU System of the RPi because the LivePhotos are quite big.

**Reply by paddywwoof** [reply]
@vinyli85, Just checking out how Apple do LivePhotos and it seems they are a jpeg and a mov (using h.264 12-15fps). I don't know how those two resources are combined but imagine they would be separate files to make it easier to use in non-iPhone locations.. In which case you could show the movie using pi3d in a similar way to the pi3d_demos VideoWalk but, just as with my gif comment, you would need to change the code of PictureFrame quite a bit to get any kind of playback speed - almost certainly down-scaling the image to a 480x270 Texture.

So quite possible to do but not a beginners project. If I get some time I might play around with some ideas.
